### PR TITLE
Add Port.monitor/1 and Port.demonitor/1 functions

### PR DESCRIPTION
**Pull request by kelvinst**
Mentioned on to #4617

### CODE REVIEW

**Comment by josevalim** [comment]
Thank you. Can you please make sure they are inlined here: https://github.com/elixir-lang/elixir/blob/master/lib/elixir/src/elixir_rewrite.erl#L144

As well as add tests?

**Comment by kelvinst** [comment]
@josevalim SURE! :)

File: lib/elixir/test/elixir/port_test.exs [hunk 1]
```diff
@@ -18,4 +18,17 @@ defmodule PortTest do
...
     assert Port.info(port) == nil
   end
+
+  test "monitoring functions are inlined by the compiler" do
+    assert expand(quote(do: Port.monitor(port())), __ENV__) ==
+           quote(do: :erlang.monitor(:port, port()))
+
+    assert expand(quote(do: Port.demonitor(port())), __ENV__) ==
+           quote(do: :erlang.demonitor(port()))
+  end
```
Comment (kelvinst) [inline lib/elixir/test/elixir/port_test.exs]: @josevalim Is this enough testing for you?

File: lib/elixir/lib/port.ex [hunk 2]
```diff
@@ -246,6 +246,44 @@ defmodule Port do
...
+
+  Inlined by the compiler.
+  """
+  @spec monitor(port) :: reference
+  def monitor(port) do
+    :erlang.monitor(:port, port)
+  end
+
+  @doc """
+  Demonitors the monitor identifies by the given `reference`.
```
Comment (bschaeffer) [inline lib/elixir/lib/port.ex]: `identified by`

File: lib/elixir/test/elixir/port_test.exs [hunk 3]
```diff
@@ -18,4 +18,17 @@ defmodule PortTest do
     assert Port.info(port, :registered_name) == nil
     assert Port.info(port) == nil
   end
+
+  test "monitoring functions are inlined by the compiler" do
```
Comment (lexmag) [inline lib/elixir/test/elixir/port_test.exs]: I believe we don't test inlining.

Comment (kelvinst) [reply lib/elixir/test/elixir/port_test.exs]: Well, [you do](https://github.com/elixir-lang/elixir/blob/master/lib/elixir/test/elixir/process_test.exs#L27). But if you think it's better to test `:erlang.monitor` and `:erlang.demonitor` functionality here, I can do it also. :)

File: lib/elixir/lib/port.ex (hunk 2 above)
Comment (kelvinst) [reply lib/elixir/lib/port.ex]: Copied typo, 😬

File: lib/elixir/test/elixir/port_test.exs (hunk 3 above)
Comment (lexmag) [reply lib/elixir/test/elixir/port_test.exs]: Hmm, we do a lot of inlining and this is perhaps the only test we have so far. 🤔

File: lib/elixir/test/elixir/port_test.exs (hunk 1 above)
Comment (josevalim) [reply lib/elixir/test/elixir/port_test.exs]: I don't want to test they are inlined, that's an implementation detail. We should check they actually work. :)
You can start a port that calls `echo` with `foo` as argument.

File: lib/elixir/test/elixir/port_test.exs (hunk 3 above)
Comment (kelvinst) [reply lib/elixir/test/elixir/port_test.exs]: OK! I'll test the functionality then!

File: lib/elixir/test/elixir/port_test.exs (hunk 1 above)
Comment (kelvinst) [reply lib/elixir/test/elixir/port_test.exs]: OK, will make it!

File: lib/elixir/test/elixir/port_test.exs [hunk 4]
```diff
@@ -18,4 +18,15 @@ defmodule PortTest do
     assert Port.info(port, :registered_name) == nil
     assert Port.info(port) == nil
   end
+
+  test "monitor/1 does monitor the given port" do
```
Comment (kelvinst) [inline lib/elixir/test/elixir/port_test.exs]: @josevalim and @lexmag now it's testing if it actually works!

**Comment by lexmag** [comment]
Thank you @kelvinst. 💛

**Pull request merged**